pkg/defaults/cloudprovider.go

```diff
@@ -0,0 +1,148 @@
+package defaults
+
+import (
+	"io"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	noProxyTransport *http.Transport
+)
+
+func init() {
+	noProxyTransport = http.DefaultTransport.(*http.Transport).Clone()
+	noProxyTransport.Proxy = nil // no proxy
+}
+
+// TryDiscoverPublicIP tries to discover the public IP of the node by querying
+// a list of known providers. If the public IP cannot be discovered, an empty
+// string is returned.
+func TryDiscoverPublicIP() string {
```

Comment:
i'd prefer this to be a simple if/else or switch statement instead

Reply:
done